api/src/main/java/org/apache/iceberg/types/CheckCompatibility.java

```diff
@@ -78,7 +78,7 @@
 
   private CheckCompatibility(Schema schema, boolean checkOrdering, boolean checkNullability) {
     this.schema = schema;
-    this.checkOrdering = checkOrdering;
+    this.checkOrdering = false;
```

Comment:
I think this should still pass `checkOrdering` correctly. To fix the problem this is trying to address, I think this should add a write option to that gets passed into this.